utils/env_preferences/src/parse/posix.rs

```diff
@@ -0,0 +1,621 @@
+        // Check if the language needs to be mapped to a BCP-47 language subtag
+        let language_code = POSIX_LANGUAGE_NAMES
+            .binary_search_by_key(&self.language, |(language, _code)| language)
+            // `binary_search_by_key()` guarantees a valid index
+            .map(|index| POSIX_LANGUAGE_NAMES[index].1);
+
+        let mut language = match language_code {
+            // Found a matching language code
+            Ok(language) => language,
+            // Parse the language directly
+            Err(_) => match Language::try_from_str(self.language) {
+                Ok(language) => language,
+                Err(_) => {
+                    return Err(ConversionError::InvalidLanguage {
+                        start_offset: 0,
+                        end_offset: self.language.len(),
+                    })
+                }
+            },
+        };
+
+        let region = if let Some(territory) = self.territory {
+            match Region::try_from_str(territory) {
+                Ok(region) => Some(region),
+                Err(_) => {
+                    return Err(ConversionError::InvalidRegion {
+                        // Add 1 to skip the delimiter
+                        start_offset: self.language.len() + 1,
+                        // Add 1 to include the final character
+                        end_offset: self.language.len() + territory.len() + 1,
+                    });
+                }
+            }
+        } else {
+            None
+        };
+
+        if let Some(codeset) = self.codeset {
+            icu_provider::log::warn!("Ignoring codeset: `{codeset}`");
+        }
+
+        let mut extensions = Extensions::new();
+        let mut script = None;
+        let mut variants = vec![variant!("posix")];
+
+        if let Some(modifier) = self.modifier {
+            match modifier.to_ascii_lowercase().as_str() {
+                "euro" => {
+                    extensions.unicode.keywords.set(key!("cu"), value!("eur"));
+                }
+                // Known script modifiers
+                "cyrillic" => script = Some(script!("Cyrl")),
+                "devanagari" => script = Some(script!("Deva")),
+                "latin" => script = Some(script!("Latn")),
+                // Saaho seems to be the only "legacy variant" that appears as a modifier:
+                // https://www.unicode.org/reports/tr35/#table-legacy-variant-mappings
+                "saaho" => {
+                    // The only known locale with this modifier is `aa_ER`
+                    if self.language != "aa" || self.territory != Some("ER") {
+                        icu_provider::log::warn!("Overriding language: `{}`->`ssy`", self.language);
+                    }
+                    language = language!("ssy")
+                }
+                // This keeps `variants` sorted; "-posix" comes before "-valencia"
+                "valencia" => variants.push(variant!("valencia")),
```

Comment:
`question`: why are you adding a variant here, and not in `-u-va`?

Reply:
For context, the [advice](https://github.com/unicode-org/icu4x/discussions/6028#discussioncomment-12200364) I received from @robertbastian:

> Similar with `@valencia`, I've seen this as both a variant and a subdivision tag (`-u-sd-valencia`). We don't currently have data keyed by either.

I could have misinterpreted the meaning of "variant" in the text above, happy to change it to `-u-va-valencia` or `-u-sd-valnecia` if that would be more appropriate!